plugins/node/opentelemetry-koa-instrumentation/src/koa.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BasePlugin } from '@opentelemetry/core';
+import * as koa from 'koa';
+import * as shimmer from 'shimmer';
+import {
+  Parameters,
+  KoaMiddleware,
+  KoaContext,
+  KoaComponentName,
+  kLayerPatched,
+} from './types';
+import { VERSION } from './version';
+import { getMiddlewareMetadata } from './utils';
+
+/** Koa instrumentation for OpenTelemetry */
+export class KoaInstrumentation extends BasePlugin<typeof koa> {
+  static readonly component = KoaComponentName;
+  readonly supportedVersions = ['^2.0.0'];
+
+  constructor(readonly moduleName: string) {
+    super('@opentelemetry/koa-instrumentation', VERSION);
+  }
+
+  /**
+   * Patches Koa operations by wrapping the Koa.use function
+   */
+  protected patch(): typeof koa {
+    this._logger.debug('Patching Koa');
+
+    if (this._moduleExports === undefined || this._moduleExports === null) {
```

Comment:
```suggestion
    if (this._moduleExports == null) {
```

Reply:
We really need to find a consensus on loose comparison :/

Reply:
For this case, should I apply the suggested changes?

Reply:
I think so. You can create a separate tracking issue off of @vmarchaud's comment

Reply:
yeah go for it, there already is an issue open there https://github.com/open-telemetry/opentelemetry-js/issues/1313